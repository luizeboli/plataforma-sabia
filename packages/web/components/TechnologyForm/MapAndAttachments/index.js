/* eslint-disable react-hooks/exhaustive-deps */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-use-before-define */
import React, { useRef, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { FaTrash, FaFileUpload, FaFilePdf, FaMapMarkerAlt } from 'react-icons/fa';
import Dropzone from 'react-dropzone';
import PlacesAutocomplete, { geocodeByPlaceId } from 'react-places-autocomplete';
import { upload, deleteUpload } from '../../../services/uploads';
import { createTerm } from '../../../services/terms';
import { InputField, SelectField, InputHiddenField } from '../../Form';
import {
	UploadedImages,
	UploadedDocuments,
	SvgBox,
	IconBox,
	IconTextBox,
	UploadBox,
	Suggestion,
	GoogleAddressSugestions,
	Div,
	IconLink,
	Wrapper,
	IconRow,
	Title,
	Media,
} from './styles';
import { Row, Column } from '../../Common/Layout';
import { CircularButton } from '../../Button';

const parseMetaObjectIntoKeyValue = (findTerm, terms) => {
	const filteredTerms = terms.filter(({ term }) => term === findTerm);

	const newTerms = filteredTerms.map(({ metas, id }) => {
		const parsedObject = {
			id,
		};
		metas.forEach((meta) => {
			parsedObject[meta.meta_key] = meta.meta_value;
		});
		return parsedObject;
	});

	return newTerms;
};

const MapAndAttachments = ({ form, data }) => {
	const { attachments, rawTerms: terms } = data.technology;
	const imgsRef = useRef(null);
	const pdfRef = useRef(null);
	const [whereIsAlreadyImplemented, setWhereIsAlreadyImplemented] = useState([]);
	const [whereIsAlreadyImplementedInput, setWhereIsAlreadyImplementedInput] = useState('');
	const [whoDevelop, setWhoDevelop] = useState([]);
	const [whoDevelopInput, setWhoDevelopInput] = useState('');
	const [previewedImgFiles, setPreviewedImgFiles] = useState(attachments.images);
	const [previewedPdfFiles, setPreviewedPdfFiles] = useState(attachments.documents);

	useEffect(() => {
		const whereIsAlreadyImplementedParsed = parseMetaObjectIntoKeyValue(
			'where_is_already_implemented',
			terms,
		);
		setWhereIsAlreadyImplemented(whereIsAlreadyImplementedParsed);

		const whoDevelopParsed = parseMetaObjectIntoKeyValue('who_develop', terms);
		setWhoDevelop(whoDevelopParsed);
	}, []);

	useEffect(() => {
		whoDevelop.forEach((element, index) => {
			form.setValue(`terms.who_develop[${index}]`, element.id);
		});
	}, [whoDevelop]);

	useEffect(() => {
		whereIsAlreadyImplemented.forEach((element, index) => {
			form.setValue(`terms.where_is_already_implemented[${index}]`, element.id);
		});
	}, [whereIsAlreadyImplemented]);

	// eslint-disable-next-line consistent-return
	const onDropAttachments = async (acceptedFiles, type) => {
		if (!acceptedFiles) return null;

		const formData = new FormData();
		for (let index = 0; index < acceptedFiles.length; index += 1) {
			formData.append(`files[${index}]`, acceptedFiles[index], acceptedFiles[index].name);
		}

		formData.append(
			'meta',
			JSON.stringify({
				object: 'technologies',
				object_id: form.getValues('id'),
			}),
		);

		const response = await upload(formData);
		if (type === 'img') {
			const newValue = [...previewedImgFiles, ...response];
			setPreviewedImgFiles(newValue);
		}

		if (type === 'pdf') {
			const newValue = [...previewedPdfFiles, ...response];
			setPreviewedPdfFiles(newValue);
		}
	};

	const deleteAttachment = async ({ index, element, type }) => {
		await deleteUpload(element.id);
		if (type === 'img') {
			setPreviewedImgFiles(
				previewedImgFiles.filter((media, innerIndex) => index !== innerIndex),
			);
		}
		if (type === 'pdf') {
			setPreviewedPdfFiles(
				previewedPdfFiles.filter((pdf, innerIndex) => index !== innerIndex),
			);
		}
	};

	// eslint-disable-next-line consistent-return
	const deleteFromCollection = (index, collection) => {
		if (collection === 'whoDevelop') {
			return setWhoDevelop(whoDevelop.filter((element, innerIndex) => index !== innerIndex));
		}

		if (collection === 'whereIsAlreadyImplemented') {
			return setWhereIsAlreadyImplemented(
				whereIsAlreadyImplemented.filter((element, innerIndex) => index !== innerIndex),
			);
		}
	};

	const onSelect = async (value, options, properties, term) => {
		const state =
			term === 'where_is_already_implemented' ? whereIsAlreadyImplemented : whoDevelop;
		const toBePushed = properties;
		if (state.some((element) => element.placeId === toBePushed.placeId)) {
			return;
		}
		const response = await geocodeByPlaceId(toBePushed.placeId);
		if (response) {
			toBePushed.location = {
				lat: response[0].geometry.location.lat(),
				lng: response[0].geometry.location.lng(),
			};
		}
		const createResponse = await createTerm(term, 'GOOGLE_PLACE', [
			{
				meta_key: 'placeId',
				meta_value: toBePushed.placeId,
			},
			{
				meta_key: 'description',
				meta_value: toBePushed.description,
			},
			{
				meta_key: 'latitude',
				meta_value: `${toBePushed.location.la}`,
			},
			{
				meta_key: 'longitude',
				meta_value: `${toBePushed.location.lng}`,
			},
		]);

		const newState = [...state, { ...toBePushed, id: createResponse.id }];
		if (term === 'where_is_already_implemented') {
			setWhereIsAlreadyImplemented(newState);
			setWhereIsAlreadyImplementedInput('');
		} else {
			setWhoDevelop(newState);
			setWhoDevelopInput('');
		}
	};

	return (
		<Wrapper>
			<Row>
				<Column>
					<Row>
						<Column>
							<Title>Onde a Tecnologia é desenvolvida?</Title>
						</Column>
					</Row>
					<Row>
						<Column>
							<PlacesAutocomplete
								id="who_develop"
								name="who_develop"
								value={whoDevelopInput}
								onChange={(value) => setWhoDevelopInput(value)}
								onSelect={(value, options, properties) =>
									onSelect(value, options, properties, 'who_develop')
								}
							>
								{({
									getInputProps,
									suggestions,
									getSuggestionItemProps,
									loading,
								}) => (
									<div>
										<InputField
											{...getInputProps({
												placeholder: 'Procurar localidades...',
												className: 'location-search-input',
											})}
											form={{ register: () => {} }}
										/>
										<div className="autocomplete-dropdown-container">
											{loading && (
												<GoogleAddressSugestions>
													Loading...
												</GoogleAddressSugestions>
											)}
											<GoogleAddressSugestions>
												{suggestions.map((suggestion) => {
													const style = suggestion.active
														? {
																backgroundColor: '#fafafa',
														  }
														: {
																backgroundColor: '#fff',
														  };
													return (
														<div
															key={suggestion.placeId}
															{...getSuggestionItemProps(suggestion, {
																style,
															})}
														>
															<Suggestion>
																{suggestion.description}
															</Suggestion>
														</div>
													);
												})}
											</GoogleAddressSugestions>
										</div>
									</div>
								)}
							</PlacesAutocomplete>
						</Column>
					</Row>
					<Row>
						<Column>
							{whoDevelop.map((element, index) => {
								return (
									<IconRow row>
										<IconLink>
											<FaMapMarkerAlt size="2rem" /> {element.description}
										</IconLink>
										<CircularButton
											variant="remove"
											height="2"
											width="2"
											onClick={() =>
												deleteFromCollection(index, 'whoDevelop')
											}
										>
											<FaTrash size="1em" />
										</CircularButton>
										<InputHiddenField
											form={form}
											type="hidden"
											ref={form.register()}
											name={`terms.who_develop[${index}]`}
										/>
									</IconRow>
								);
							})}
						</Column>
					</Row>
					<Row>
						<Column>
							<Title>Onde a Tecnologia pode ser aplicada?</Title>
						</Column>
					</Row>
					<Row>
						<Column>
							<SelectField
								form={form}
								id="where_can_be_applied"
								label="Região"
								name="terms.where_can_be_applied"
								placeholder="Escolha uma região"
								validation={{ required: true }}
								options={[
									{
										value: 'semiarido',
										label: 'Semiárido',
									},
								]}
							/>
						</Column>
					</Row>

					<Row>
						<Column>
							<Title>Onde a Tecnologia já está implantada?</Title>
						</Column>
					</Row>
					<Row>
						<Column>
							<PlacesAutocomplete
								id="where_is_already_implemented"
								name="where_is_already_implemented"
								value={whereIsAlreadyImplementedInput}
								onChange={(value) => setWhereIsAlreadyImplementedInput(value)}
								onSelect={(value, options, properties) =>
									onSelect(
										value,
										options,
										properties,
										'where_is_already_implemented',
									)
								}
							>
								{({
									getInputProps,
									suggestions,
									getSuggestionItemProps,
									loading,
								}) => (
									<div>
										<InputField
											{...getInputProps({
												placeholder: 'Procurar localidades...',
												className: 'location-search-input',
											})}
											form={{ register: () => {} }}
										/>
										<div className="autocomplete-dropdown-container">
											{loading && (
												<GoogleAddressSugestions>
													Loading...
												</GoogleAddressSugestions>
											)}
											<GoogleAddressSugestions>
												{suggestions.map((suggestion) => {
													const style = suggestion.active
														? {
																backgroundColor: '#fafafa',
														  }
														: {
																backgroundColor: '#fff',
														  };
													return (
														<div
															{...getSuggestionItemProps(suggestion, {
																style,
															})}
															key={suggestion.placeId}
														>
															<Suggestion>
																{suggestion.description}
															</Suggestion>
														</div>
													);
												})}
											</GoogleAddressSugestions>
										</div>
									</div>
								)}
							</PlacesAutocomplete>
						</Column>
					</Row>
					<Row>
						<Column>
							{whereIsAlreadyImplemented.map((element, index) => {
								return (
									<IconRow row>
										<IconLink>
											<FaMapMarkerAlt size="2rem" /> {element.description}
										</IconLink>
										<CircularButton
											variant="remove"
											height="2"
											width="2"
											onClick={() =>
												deleteFromCollection(
													index,
													'whereIsAlreadyImplemented',
												)
											}
										>
											<FaTrash size="1em" />
										</CircularButton>
										<InputHiddenField
											form={form}
											type="hidden"
											ref={form.register()}
											name={`terms.where_is_already_implemented[${index}]`}
										/>
									</IconRow>
								);
							})}
						</Column>
					</Row>
				</Column>
				<Column>
					<Title>Fotos e Vídeos da tecnologia</Title>

					<Dropzone
						accept="image/*"
						onDrop={(acceptedFiles) => onDropAttachments(acceptedFiles, 'img')}
						ref={imgsRef}
					>
						{({ getRootProps, getInputProps }) => (
							<UploadBox {...getRootProps()}>
								<input {...getInputProps()} />
								<SvgBox>
									<IconBox>
										<FaFileUpload size="5rem" />
										<IconTextBox>
											<Div>Upload</Div>
											<Div>1280px x 720px</Div>
										</IconTextBox>
									</IconBox>
								</SvgBox>
							</UploadBox>
						)}
					</Dropzone>
					<UploadedImages>
						{previewedImgFiles?.map((element, index) => {
							return (
								<IconRow key={element.url}>
									<Media key={element.url} src={element.url} />
									<CircularButton
										variant="remove"
										height="3"
										width="3"
										onClick={() =>
											deleteAttachment({
												index,
												element,
												type: 'img',
											})
										}
									>
										<FaTrash size="1.5em" />
									</CircularButton>
								</IconRow>
							);
						})}
					</UploadedImages>

					<Title>Documentos</Title>
					<Dropzone
						accept=".pdf"
						onDrop={(acceptedFiles) => onDropAttachments(acceptedFiles, 'pdf')}
						ref={pdfRef}
					>
						{({ getRootProps, getInputProps }) => (
							<UploadBox {...getRootProps()}>
								<input {...getInputProps()} />
								<SvgBox>
									<IconBox>
										<FaFileUpload size="5rem" />
										<IconTextBox>
											<Div>Upload</Div>
											<Div>.PDF</Div>
										</IconTextBox>
									</IconBox>
								</SvgBox>
							</UploadBox>
						)}
					</Dropzone>
					<UploadedDocuments>
						{previewedPdfFiles?.map((element, index) => {
							return (
								<IconRow row>
									<IconLink href={element.url}>
										<FaFilePdf size="2rem" /> {element.filename}
									</IconLink>
									<CircularButton
										variant="remove"
										height="2"
										width="2"
										onClick={() =>
											deleteAttachment({
												index,
												element,
												type: 'pdf',
											})
										}
									>
										<FaTrash size="1em" />
									</CircularButton>
								</IconRow>
							);
						})}
					</UploadedDocuments>
				</Column>
			</Row>
		</Wrapper>
	);
};

MapAndAttachments.propTypes = {
	form: PropTypes.shape({
		getValues: PropTypes.func,
		register: PropTypes.func,
		setValue: PropTypes.func,
	}),
	data: PropTypes.shape({
		technology: PropTypes.shape({
			attachments: PropTypes.arrayOf(PropTypes.shape({})),
			rawTerms: PropTypes.arrayOf(PropTypes.shape({})),
		}),
	}),
};

MapAndAttachments.defaultProps = {
	form: {},
	data: {
		technology: {
			attachments: [],
			rawTerms: [],
		},
	},
};

export default MapAndAttachments;
