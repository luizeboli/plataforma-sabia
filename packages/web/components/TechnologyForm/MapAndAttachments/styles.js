import styled from 'styled-components';

export const Wrapper = styled.div`
	margin-bottom: 4rem;
`;

export const UploadBox = styled.div`
	flex-direction: row;
	display: flex;
	margin: 0;
	margin-bottom: 1rem;
	border: 1px solid ${({ theme }) => theme.colors.lightGray3};
	padding: 1rem;
	cursor: pointer;

	&:hover {
		background-color: ${({ theme }) => theme.colors.lightGray3};
	}
`;

export const IconLink = styled.a`
	color: ${({ theme }) => theme.colors.darkGray};
`;

export const Place = styled.div``;

export const Div = styled.div``;

export const IconBox = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;

	& > * {
		margin: 0.25rem;
	}
`;

export const IconTextBox = styled.div`
	font-size: 1.2rem;
	font-weight: 300;
	display: flex;
	justify-content: flex-start;
	align-items: flex-start;
	flex-direction: column;
`;

export const InputTextBox = styled.div`
	flex: 3;
	margin: auto 0;
`;

export const UploadedDocuments = styled.div`
	display: flex;
	flex-direction: column;
`;

export const UploadedImages = styled.div`
	display: grid;
	grid-template-columns: repeat(auto-fill, 30%);
	grid-gap: 1rem;

	@media (max-width: ${({ theme }) => theme.screens.small}px) {
		grid-template-columns: repeat(auto-fill, 100%);
	}
`;

export const Suggestion = styled.div`
	margin: 0.5rem 0;
	padding: 1rem;
	width: 100%;
	font-size: 1.4rem;
	cursor: pointer:
`;

export const GoogleAddressSugestions = styled.div`
	display: flex;
	flex-direction: column;
`;

export const SvgBox = styled.div`
	text-align: center;
	flex: 1;
	margin: auto 0;
`;

export const Title = styled.h3``;

export const Media = styled.img`
	width: 100%;
	filter: opacity(0.7);
`;

export const IconRow = styled.div`
	display: flex;
	flex-direction: ${({ row }) => (row ? 'row' : 'column')};
	align-items: center;

	& > * {
		margin-bottom: 1rem;
	}

	& > a {
		padding: 0;
		margin-right: 1rem;
		display: flex;
	}
`;
