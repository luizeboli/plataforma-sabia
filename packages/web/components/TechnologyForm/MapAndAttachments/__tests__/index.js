import React from 'react';
import { render } from 'test-utils';
import MapAndAttachments from '..';
import { Form } from '../../../Form';

jest.mock('react-places-autocomplete', () => {
	return {
		__esModule: true,
		geocodeByPlaceId: () => {
			return <vid>geocodeByPlaceId</vid>;
		},
		default: () => {
			return <div>PlacesAutocomplete</div>;
		},
	};
});

const onSubmit = jest.fn(() => {});

test('it render the MapAndAttachments page', () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<MapAndAttachments />
		</Form>,
	);

	expect(container).toMatchSnapshot();
});
