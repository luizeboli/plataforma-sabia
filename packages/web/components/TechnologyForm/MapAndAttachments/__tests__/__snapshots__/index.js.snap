// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it render the MapAndAttachments page 1`] = `
.c12 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c13 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c14 {
  width: 100%;
  margin: 1rem 0;
  font-size: 1.4rem;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: hsla(195,86%,64%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  left: 5px;
}

.c11:hover {
  opacity: 0.8;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: hsl(0,0%,74%);
  color: hsla(0,0%,100%);
  border-radius: 100%;
  height: 100%;
  border: none;
  font-size: 1.2rem;
  text-transform: uppercase;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0.2rem !important;
  float: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  left: 5px;
}

.c10:hover {
  opacity: 0.8;
}

.c1 {
  margin-bottom: 4rem;
}

.c15 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  margin-bottom: 1rem;
  border: 1px solid hsl(0,0%,74%);
  padding: 1rem;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c17 > * {
  margin: 0.25rem;
}

.c18 {
  font-size: 1.2rem;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c19 {
  display: grid;
  grid-template-columns: repeat(auto-fill,30%);
  grid-gap: 1rem;
}

.c16 {
  text-align: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: auto 0;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0 1rem;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c9 {
  margin: 0 auto;
  padding: 0 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: 0rem;
  margin-bottom: 1rem;
  background-color: transparent;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 0rem;
  margin-bottom: 1rem;
  background-color: transparent;
}

.c4 {
  margin-bottom: 3rem;
}

.c6 {
  margin-top: 2rem;
}

@media (max-width:450px) {
  .c19 {
    grid-template-columns: repeat(auto-fill,100%);
  }
}

@media (max-width:1170px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media (max-width:1170px) {
  .c7 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <h3
                class=""
              >
                Onde a Tecnologia é desenvolvida?
              </h3>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <div
                class="c4"
              >
                <div
                  class="c5"
                />
                <div
                  class="c6"
                >
                  <div
                    class="c7"
                  >
                    <div
                      class="c8"
                    >
                      <div
                        class=" css-2b097c-container"
                      >
                        <div
                          class=" css-yk16xz-control"
                        >
                          <div
                            class=" css-g1d714-ValueContainer"
                          >
                            <div
                              class=" css-1wa3eu0-placeholder"
                            >
                              Nome da instituição que desenvolve a tecnologia
                            </div>
                            <div
                              class="css-b8ldur-Input"
                            >
                              <div
                                class=""
                                style="display: inline-block;"
                              >
                                <input
                                  aria-autocomplete="list"
                                  autocapitalize="none"
                                  autocomplete="off"
                                  autocorrect="off"
                                  id="react-select-3-input"
                                  spellcheck="false"
                                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                                  tabindex="0"
                                  type="text"
                                  value=""
                                />
                                <div
                                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                                />
                              </div>
                            </div>
                          </div>
                          <div
                            class=" css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              class=" css-1okebmr-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              class=" css-tlfecz-indicatorContainer"
                            >
                              <svg
                                aria-hidden="true"
                                class="css-6q0nyr-Svg"
                                focusable="false"
                                height="20"
                                viewBox="0 0 20 20"
                                width="20"
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                        <input
                          name="who_develop[0]"
                          type="hidden"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="c9"
                    >
                      <button
                        class="c10"
                        color="hsla(0, 0%, 100%)"
                        disabled=""
                        name="who_develop[0]_remove_button"
                        title=""
                        type="button"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 448 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                          />
                        </svg>
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="c2"
                >
                  <div
                    class="c3"
                  >
                    <button
                      class="c11"
                      color="hsla(0, 0%, 100%)"
                      name="who_develop_add_button"
                      title=""
                      type="button"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 448 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <h3
                class=""
              >
                Onde a Tecnologia pode ser aplicada?
              </h3>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <div
                class="c12"
              >
                <label
                  class="c13"
                  for="where_can_be_applied"
                >
                  Região
                </label>
                <div
                  class="c5"
                >
                  <div
                    class="c14 react-select-container css-2b097c-container"
                    id="where_can_be_applied"
                  >
                    <div
                      class="react-select__control css-yk16xz-control"
                    >
                      <div
                        class="react-select__value-container css-g1d714-ValueContainer"
                      >
                        <div
                          class="react-select__placeholder css-1wa3eu0-placeholder"
                        >
                          Escolha uma região
                        </div>
                        <div
                          class="css-b8ldur-Input"
                        >
                          <div
                            class="react-select__input"
                            style="display: inline-block;"
                          >
                            <input
                              aria-autocomplete="list"
                              aria-label="Região"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              id="react-select-2-input"
                              spellcheck="false"
                              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="react-select__indicator-separator css-1okebmr-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-6q0nyr-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="where_can_be_applied"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <h3
                class=""
              >
                Onde a Tecnologia já está implantada?
              </h3>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            >
              <div>
                PlacesAutocomplete
              </div>
            </div>
          </div>
          <div
            class="c2"
          >
            <div
              class="c3"
            />
          </div>
        </div>
        <div
          class="c3"
        >
          <h3
            class=""
          >
            Fotos e Vídeos da tecnologia
          </h3>
          <div
            class="c15"
            tabindex="0"
          >
            <input
              accept="image/*"
              autocomplete="off"
              multiple=""
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <div
              class="c16"
            >
              <div
                class="c17"
              >
                <svg
                  fill="currentColor"
                  height="5rem"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 384 512"
                  width="5rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z"
                  />
                </svg>
                <div
                  class="c18"
                >
                  <div
                    class=""
                  >
                    Upload
                  </div>
                  <div
                    class=""
                  >
                    1280px x 720px
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c19"
          />
          <h3
            class=""
          >
            Documentos
          </h3>
          <div
            class="c15"
            tabindex="0"
          >
            <input
              accept=".pdf"
              autocomplete="off"
              multiple=""
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <div
              class="c16"
            >
              <div
                class="c17"
              >
                <svg
                  fill="currentColor"
                  height="5rem"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 384 512"
                  width="5rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm65.18 216.01H224v80c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16v-80H94.82c-14.28 0-21.41-17.29-11.27-27.36l96.42-95.7c6.65-6.61 17.39-6.61 24.04 0l96.42 95.7c10.15 10.07 3.03 27.36-11.25 27.36zM377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9z"
                  />
                </svg>
                <div
                  class="c18"
                >
                  <div
                    class=""
                  >
                    Upload
                  </div>
                  <div
                    class=""
                  >
                    .PDF
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c20"
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
